tests/tpch/test_dask.py

```diff
@@ -519,3 +519,81 @@ def test_query_12(client, dataset_path, fs):
         .sort_values(by="l_shipmode")
         .compute()
     )
+
+
+@pytest.mark.shuffle_p2p
+@pytest.mark.xfail(
+    raises=RuntimeError,
+    reason=(
+        "Works w/ dask.dataframe, dask-expr fails - "
+        "RuntimeError: Sequence [('dtype_backend', None)] "
+        "cannot be deterministically hashed."
+    ),
+)
+def test_query_16(client, dataset_path, fs):
+    """
+    select
+        p_brand,
+        p_type,
+        p_size,
+        count(distinct ps_suppkey) as supplier_cnt
+    from
+        partsupp,
+        part
+    where
+        p_partkey = ps_partkey
+        and p_brand <> 'Brand#45'
+        and p_type not like 'MEDIUM POLISHED%'
+        and p_size in (49, 14, 23, 45, 19, 3, 36, 9)
+        and ps_suppkey not in (
+            select
+                s_suppkey
+            from
+                supplier
+            where
+                s_comment like '%Customer%Complaints%'
+        )
+    group by
+        p_brand,
+        p_type,
+        p_size
+    order by
+        supplier_cnt desc,
+        p_brand,
+        p_type,
+        p_size
+    """
+    partsupp = dd.read_parquet(dataset_path + "partsupp", filesystem=fs)
+    part = dd.read_parquet(dataset_path + "part", filesystem=fs)
+    supplier = dd.read_parquet(dataset_path + "supplier", filesystem=fs)
+
+    table = partsupp.merge(
+        part, left_on="ps_partkey", right_on="p_partkey", how="inner"
+    )
+    table = table[
+        (table.p_brand != "Brand#45")
+        & (~table.p_type.str.match("MEDIUM POLISHED*"))
+        & (table.p_size.isin((49, 14, 23, 45, 19, 3, 36, 9)))
+        & (
+            # Need to compute, otherwise get NotImplementedErrror
+            # this works for dask.dataframe, fails for dask-expr w/
+            # RuntimeError: Sequence [('dtype_backend', None)] cannot be deterministically hashed.
+            ~table.ps_suppkey.isin(
+                supplier[
+                    supplier.s_comment.str.match("*Customer*Complaints*")
+                ].s_suppkey.compute()
```

Comment:
If this works for dask.dataframe without calling compute, I'd suggest to remove the compute, XFAIL the test and file an issue for `dask-expr`.

Reply:
Already xfailed.. it needs a compute, otherwise it fails in both with NotImplementedError. 
https://github.com/dask-contrib/dask-expr/issues/805

Reply:
actually, did a small refactor and it works for both dask.dataframe and dask-expr again.